frontend/src/metabase/visualizations/visualizations/PieChart/PieChart.jsx

```diff
@@ -258,11 +261,15 @@ export default class PieChart extends Component {
       const { width, height } =
         this.chartContainer.current.getBoundingClientRect();
 
+      if (width === 0 && height === 0) {
+        return this.updateChartViewportSize(delay + 200);
```

Comment:
This looks a bit hackish, it would be great to avoid that.
So the problem is that ExplicitSize in `ChartWithLegend` resizes with a delay so that this `updateChartViewportSize` function is invoked when chartContainer still has zero width and zero height.

I think a more straightforward and reliable approach would be to call the previous version of `updateChartViewportSize` when ExplicitSize updates its size. Wdyt?

Reply:
Thank you for the feedback! I pushed up a commit to revert the hack and update the chart viewport size when ExplicitSize updates its size.